lib/repl.js

```diff
@@ -763,6 +763,8 @@ exports.start = function(prompt,
                          useGlobal,
                          ignoreUndefined,
                          replMode) {
+  console.log(`Welcome to Node.js ${process.version}.\n` +
+    'Type ".help" for more information.');
```

Comment:
This is not going to be visible in the REPL instance itself but in the terminal which started the REPL / the REPL starting Node.js instance. Instead, please write to the output stream of the created instance instead.

Reply:
`console.log` can not be used here at it does not know anything about the REPL. It has to be similar to https://github.com/nodejs/node/blob/9bbe29dcce8ee7830e5589a407e56e9cfcfc2129/lib/repl.js#L631 and it should be added at the end of the instance constructor / right before the prompt call here: https://github.com/nodejs/node/blob/9bbe29dcce8ee7830e5589a407e56e9cfcfc2129/lib/repl.js#L748 :)

Reply:
Actually the code do works in this way. I apply the change you suggested, but the build failed 18 test cases.
See https://travis-ci.org/gengjiawen/node/jobs/489123322. I reverted it to console for now.